src/core/engines/Cesium/Feature/index.tsx

```diff
@@ -70,56 +73,85 @@ const pickProperty = (k: keyof AppearanceTypes, layer: ComputedLayer) => {
   return layer.layer[k];
 };
 
+const CACHED_COMPONENTS = new LRUCache<string, JSX.Element>({ max: 10000 });
+const FEATURE_DELEGATE_THRESHOLD = 6000;
+
 export default function Feature({
   layer,
   isHidden,
   ...props
 }: FeatureComponentProps): JSX.Element | null {
+  // Set the displayConfig based on the size of the feature collection array
+  if (layer.features?.length > FEATURE_DELEGATE_THRESHOLD) {
+    displayConfig.geojson = ["resource"];
+  } else {
+    displayConfig.geojson = "auto";
+  }
```

Comment:
I think it shouldn't override global variable, so we should define other local variable for this case.

Reply:
Good Catch. Fixed.